azuredevops/internal/service/permissions/resource_project_permissions_test.go

```diff
@@ -0,0 +1,51 @@
+// +build all permissions resource_project_permissions
+// +build !exclude_permissions !resource_project_permissions
+
+package permissions
+
+// The tests in this file use the mock clients in mock_client.go to mock out
+// the Azure DevOps client operations.
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/stretchr/testify/assert"
+)
+
+func init() {
+	/* add code for test setup here */
+}
+
+/**
+ * Begin unit tests
+ */
+
+var projectID = "9083e944-8e9e-405e-960a-c80180aa71e6"
+var projectToken = fmt.Sprintf("$PROJECT:vstfs:///Classification/TeamProject/%s", projectID)
+
+func getProjecPermissionsResource(t *testing.T, projectID string) *schema.ResourceData {
+	d := schema.TestResourceDataRaw(t, ResourceProjectPermissions().Schema, nil)
+	if projectID != "" {
+		d.Set("project_id", projectID)
+	}
+	return d
+}
+
+func TestProjectPermissions_CreateProjectToken(t *testing.T) {
```

Comment:
Super nit: Can you switch the order of these functions, so that the public functions are written first?

Reply:
Done